cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/ProgramStatusTrigger.java

```diff
@@ -45,11 +53,71 @@ public ProgramStatusTrigger(ProgramId programId, ProgramStatus... programStatuse
 
   @Override
   public boolean isSatisfied(List<Notification> notifications) {
-    return true;
```

Comment:
no, it's just moving the logic to `ProgramStatusTrigger` from the subscriber service and handle the case of `ProgramStatusTrigger` in composite trigger